- first useful open-api/v2/hafven.yaml API

muenalan opened the pull request.

muenalan committed:
- first useful open-api/v2/hafven.yaml API

muenalan committed:
- added nodejs/javascript client (Generated with https://editor.swagger.io)

terribleherbst commented on open-api/clients/javascript-client/README.md:
```diff
@@ -0,0 +1,179 @@
+# havfen_community_api
+
+HavfenCommunityApi - JavaScript client for havfen_community_api
```
Here are some small typos. It should be called Hafven, there is a mix of havfen and hafven in the document. Guess we should call it hafven.

terribleherbst commented on open-api/clients/javascript-client/.travis.yml:
```diff
@@ -0,0 +1,7 @@
+language: node_js
```
Do we cirlcle ci or travis for continous integration? 
Which system does which part?

terribleherbst commented on open-api/clients/javascript-client/package.json:
```diff
@@ -0,0 +1,21 @@
+{
+  "name": "havfen_community_api",
+  "version": "0.0.1",
+  "description": "Hafven_Community_API__Development__This_might_link_to_the_homepage__httphafven_community_ngrok_io_httphafven_community_ngrok_io__Notice_that_all_html_tags_are_valid_in_this_description__Feel_free_to_be_descriptive_",
```
As mentioned in slack, we sould fix URL information and license

terribleherbst commented on open-api/clients/javascript-client/src/api/CommunityApi.js:
```diff
@@ -0,0 +1,282 @@
+ * OpenAPI spec version: 0.0.1
+ * Contact: apiteam@hafven-community.ngrok.io
+ *
+ * NOTE: This class is auto generated by the swagger code generator program.
```
is there any option generation this class as part of CI instead of using git for storing generated code?

muenalan replied:
@terribleherbst Great idea! Two ways:

For a nice [tutorial](https://blog.cloudboost.io/adding-swagger-to-existing-node-js-project-92a6624b855b)


a) "on-the-fly solutions": no code-generation is required. There are many npm modules helping there. Say [swagger-node](https://github.com/swagger-api/swagger-node), or likely better maintained [swagger-ui-express](https://www.npmjs.com/package/swagger-ui-express). 


b) Do the codegen in the circleCI environment, say using the original package, [swagger-node-codegen](https://www.npmjs.com/package/swagger-node-codegen), or in a 'ready-to-use' dockerhub [swagger-generator](https://hub.docker.com/r/swaggerapi/swagger-generator/)  codegen image.

terribleherbst replied:
Maybe we should include this information somewhere in the project to make sure updating the API will also update the swagger documentation.

muenalan replied:
Feel free to add a Readme.md under open-api ;)

muenalan committed:
Fixed typos

muenalan committed:
Added openapi -> serverless server, moved some dirs

	new file:   servers/serverless/README.md
	new file:   servers/serverless/output-generated/community.js
	new file:   servers/serverless/output-generated/groups.js
	new file:   servers/serverless/output-generated/handler.js
	new file:   servers/serverless/output-generated/serverless.yml
	new file:   servers/serverless/output-generated/user.js
	new file:   servers/serverless/package-lock.json
	new file:   servers/serverless/package.json
	new file:   servers/serverless/serverless-head.yml

muenalan commented:
Have added openapi -> serverless generation. New directory structure:

open-api/v2/hafven.yaml
open-api/v2/servers/serverless                                   - npm install & npm run generate 
open-api/v2/servers/serverless/output-generated    - ready to use serverless deployment
open-api/v2/clients/..

muenalan committed:
openapi-serverless: 'npm run combine' will take the root serverless.yml as a start

muenalan committed:
README.md updated
